dataprofiler/tests/data_readers/test_json_data.py

```diff
@@ -45,50 +45,66 @@ def setUpClass(cls):
             dict(path=os.path.join(test_dir, "json/honeypot_intentially_mislabeled_file.csv"), encoding='utf-8', count=14),
             dict(path=os.path.join(test_dir, "json/honeypot_intentially_mislabeled_file.parquet"), encoding='utf-8', count=14)
         ]
-    
-    def test_is_match_for_string_streams(self):
-        """
-        Determine if the json file can be automatically identified from
-        string stream
-        """
-        for input_file in self.input_file_names:
-            print(input_file)
-            with open(input_file['path'], 'r',
-                      encoding=input_file['encoding']) as fp:
-                byte_string = StringIO(fp.read())
-                self.assertTrue(JSONData.is_match(byte_string))
-
-    def test_is_match_for_byte_streams(self):
+
+        cls.buffer_list = []
+        for input_file in cls.input_file_names:
+            # add StringIO
+            buffer_info = input_file.copy()
+            with open(input_file['path'], 'r', encoding=input_file['encoding']) as fp:
+                buffer_info['path'] = StringIO(fp.read())
+            cls.buffer_list.append(buffer_info)
+            
+            # add BytesIO
+            buffer_info = input_file.copy()
+            with open(input_file['path'], 'rb') as fp:
+                buffer_info['path'] = BytesIO(fp.read())
+            cls.buffer_list.append(buffer_info)
+
+        cls.file_or_buf_list = cls.input_file_names + cls.buffer_list
+
+    @classmethod
+    def setUp(cls):
+        for buffer in cls.buffer_list:
+            buffer['path'].seek(0)
+
+    def test_is_match(self):
         """
         Determine if the json file can be automatically identified from
         byte stream
         """
-        for input_file in self.input_file_names:
-            with open(input_file['path'], 'rb') as fp:
-                byte_string = BytesIO(fp.read())
-                self.assertTrue(JSONData.is_match(byte_string))
+        for input_file in self.file_or_buf_list:
+            if isinstance(input_file, StringIO):
+                with open(input_file['path'], 'r',
+                        encoding=input_file['encoding']) as fp:
+                    byte_string = StringIO(fp.read())
+                    self.assertTrue(JSONData.is_match(byte_string))
+
+            elif isinstance(input_file, BytesIO):
+                with open(input_file['path'], 'rb') as fp:
+                    byte_string = BytesIO(fp.read())
+                    self.assertTrue(JSONData.is_match(byte_string))
```

Comment:
I think we can reduce this down to just 3 lines where we loop through and assert that is match is true for the is_match call on the object